src/app.ts

```diff
@@ -1,45 +1,59 @@
-import path from 'path';
-import logger from '../logger';
-import { User, Product } from 'models';
-import DirWatcher from 'utils/dirwatcher';
-import Importer from 'utils/importer';
-import csvConverter from 'utils/csvConverter';
-import { DIRECTORY_FILES_CHANGED } from 'constants/EventNames';
-
-const PATH_TO_DATA = path.join(__dirname, '../data');
-const WATCH_DELAY_MILLISECONDS = 2000;
-
-const user = new User();
-const product = new Product();
-const watcher = new DirWatcher();
-const importer = new Importer();
-
-watcher.watch(PATH_TO_DATA, WATCH_DELAY_MILLISECONDS);
-
-watcher.on(DIRECTORY_FILES_CHANGED, changedFiles => {
-    const asyncReadOperations = [];
-    const syncReadResults = [];
-
-    changedFiles.forEach(fileName => {
-        asyncReadOperations.push(importer.import(path.join(PATH_TO_DATA, fileName)));
-        syncReadResults.push(importer.importSync(path.join(PATH_TO_DATA, fileName)));
-    });
-
-    Promise.all(asyncReadOperations)
-        .then(data => {
-            data.forEach(csvString => {
-                csvConverter
-                    .convertToJSON(csvString)
-                    .then(csvJSON => console.log('async csvJSON------------>', csvJSON));
-            });
-        })
-        .catch(e => {
-            logger.error(e);
-        });
-
-    syncReadResults.forEach(csvString => {
-        csvConverter
-            .convertToJSON(csvString)
-            .then(csvJSON => console.log('sync csvJSON------------>', csvJSON));
-    });
-});
+import * as express from 'express';
+import * as bodyParser from 'body-parser';
+import cookieParser from 'cookie-parser';
+import errorHandler from 'errorhandler';
+import runWatchers from 'utils/runWatcher';
+import { addRoutes } from 'utils/addRoutes';
+import { queryParser } from 'middlewares/queryParser';
+
+// http servers
+import 'http-servers/plain-text-server';
+import 'http-servers/html-server';
+import 'http-servers/json-server';
+import 'http-servers/echo-server';
+
+// run file watchers from previous tasks
+runWatchers();
+
+// Made Server singleton mostly for learning purposes
+class Server {
```

Comment:
Module is a singleton by itself. Class seems here burdensome

Reply:
I like using OO style and I don't see any problems with class here. Maybe you are right about singleton here, but I'be decided to implement it for learning purposes.